connectors/src/api/webhooks/webhook_slack.ts

```diff
@@ -141,82 +147,123 @@ const _webhookSlackAPIHandler = async (
             status_code: 400,
           });
         }
-
-        if (!req.body.event?.channel) {
-          return apiError(req, res, {
-            api_error: {
-              type: "invalid_request_error",
-              message: "Missing channel in request body for message event",
+        if (req.body.event?.channel_type === "im") {
+          //Got a private message
+          if (req.body.event.subtype === "message_changed") {
+            // Ignore message_changed events in private messages
+            return res.status(200).send();
+          }
+          const slackConfig = await SlackConfiguration.findOne({
+            where: {
+              slackTeamId: req.body.team_id,
+              botEnabled: true,
             },
-            status_code: 400,
           });
-        }
+          if (!slackConfig) {
+            return apiError(req, res, {
+              api_error: {
+                type: "connector_configuration_not_found",
+                message: `Slack configuration not found for teamId ${req.body.team_id}. Are you sure the bot is not enabled?`,
+              },
+              status_code: 404,
+            });
+          }
+          const connector = await Connector.findByPk(slackConfig.connectorId);
+          if (!connector) {
+            return apiError(req, res, {
+              api_error: {
+                type: "connector_not_found",
+                message: `Connector ${slackConfig.connectorId} not found`,
+              },
+              status_code: 404,
+            });
+          }
+          const slackAccessToken = await getAccessToken(connector.connectionId);
+          const myUserId = await whoAmIMemoized(slackAccessToken);
+          if (req.body.event?.user === myUserId) {
+            // Message sent from the bot itself.
+            return res.status(200).send();
```

Comment:
curiosity: I thought the message sent by the bot was dumped via the subtype: "message_created"; why do we need both?